src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```
@@ -0,0 +1,30 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
```

Constructor does not validate the provided `operationStrategy`. The checklist requires methods/constructors to validate non-null inputs (3.10). Add a guard such as `if (operationStrategy == null) { throw new RuntimeException("OperationStrategy cannot be null"); }` before assigning the field to fail fast on wiring errors.